javaagent-tooling/src/main/java/io/opentelemetry/javaagent/tooling/ExtensionClassLoader.java

```diff
@@ -61,11 +69,13 @@ public static ClassLoader getInstance(
 
     includeEmbeddedExtensionsIfFound(extensions, javaagentFile);
 
-    extensions.addAll(parseLocation(earlyConfig.getString(EXTENSIONS_CONFIG), javaagentFile));
+    String extensionLocation = earlyConfig.getString(EXTENSIONS_CONFIG);
+    extensions.addAll(parseLocation(extensionLocation, javaagentFile));
 
     // TODO when logging is configured add warning about deprecated property
 
     if (extensions.isEmpty()) {
+      extensionClassLoadWarningMessage = "Extension file \"" + extensionLocation + "\" not found.";
```

Comment:
IIUC this will
 * never warn if there are any embedded in the agent-jar, even if `OTEL_JAVAAGENT_EXTENSIONS ` points to a bad location (because `extensions.isEmpty()` would return false)
 * warn when there are no extensions, even if `OTEL_JAVAAGENT_EXTENSIONS` is not set
 
 Therefore I'd propose to slightly alter the implementation:
```java
String externalExtensionsLocation = earlyConfig.getString(EXTENSIONS_CONFIG);
if (externalExtensionsLocation != null && !externalExtensionsLocation.isEmpty()) {
  List<URL> externalExtensions = parseLocation(externalExtensions, javaagentFile);
  extensions.addAll(externalExtensions);
  if (externalExtensions.isEmpty()) {
    extensionClassLoadWarningMessage = "No extensions found for configured location \"" + externalExtensionsLocation + "\"";
  }
}

// TODO when logging is configured add warning about deprecated property

if (extensions.isEmpty()) {
  return parent;
}
 ```

Reply:
Thanks for the advice! I will polish the log.